main.tf

```diff
@@ -15,19 +15,38 @@
  */
 
 locals {
-  seed_project_id             = format("%s-%s", var.project_prefix, "seed")
-  impersonation_apis          = distinct(concat(var.activate_apis, ["serviceusage.googleapis.com", "iamcredentials.googleapis.com"]))
-  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
-  activate_apis               = var.sa_enable_impersonation == true ? local.impersonation_apis : var.activate_apis
+  initial_activate_apis = var.project.activate_apis != null ? var.project.activate_apis : [
```

Comment:
I would prefer we keep these values as default values for the variable, it keeps the logic a little simpler and the module interface more well understood if you want to change the default values you just override the variable.